From 46dcb4d5f8f8b6411c78c13b9824b895e7fcddec Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Tue, 17 Jul 2018 12:47:39 +0800
Subject: [PATCH] arm64: dts: rockchip: enable Bluetooth PCM sound for RK3308
 EVB Boards

By default, only using lrck_tx for PCM by hardware,
therefore, we need to use I2S_CKR_TRCM_TXONLY.

Change-Id: I6c4077e7e7e65b8a3a21416fd61d5900b3b72f42
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts   | 10 ++++++++++
 arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts   | 10 ++++++++++
 .../boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts      |  9 +++++++--
 .../boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts      |  9 +++++++--
 .../boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts      |  9 +++++++--
 .../boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts      |  9 +++++++--
 6 files changed, 48 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
index 1546848258b4..a2ee26b73f73 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
@@ -23,6 +23,16 @@
 	};
 };
 
+&bluetooth_sound {
+	status = "okay";
+};
+
+&i2s_2ch_0 {
+	status = "okay";
+	#sound-dai-cells = <0>;
+	rockchip,clk-trcm = <1>;
+};
+
 &is31fl3236 {
 	reg = <0x3f>;
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts
index 30356481da9a..387497f15e39 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts
@@ -22,6 +22,16 @@
 	};
 };
 
+&bluetooth_sound {
+	status = "okay";
+};
+
+&i2s_2ch_0 {
+	status = "okay";
+	#sound-dai-cells = <0>;
+	rockchip,clk-trcm = <1>;
+};
+
 &is31fl3236 {
 	reg = <0x3f>;
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts
index 94abbecba79a..616ae24ed225 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts
@@ -58,9 +58,14 @@
 	status = "disabled";
 };
 
-&dummy_codec {
-	status = "disabled";
+&bluetooth_sound {
+	status = "okay";
+};
+
+&i2s_2ch_0 {
+	status = "okay";
 	#sound-dai-cells = <0>;
+	rockchip,clk-trcm = <1>;
 };
 
 &i2s_8ch_0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts
index e7676ef61e79..fb6602131301 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts
@@ -46,9 +46,14 @@
 	status = "disabled";
 };
 
-&dummy_codec {
-	status = "disabled";
+&bluetooth_sound {
+	status = "okay";
+};
+
+&i2s_2ch_0 {
+	status = "okay";
 	#sound-dai-cells = <0>;
+	rockchip,clk-trcm = <1>;
 };
 
 &i2s_8ch_0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts
index 7a2e837138ca..acd306428c49 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts
@@ -45,9 +45,14 @@
 	status = "disabled";
 };
 
-&dummy_codec {
-	status = "disabled";
+&bluetooth_sound {
+	status = "okay";
+};
+
+&i2s_2ch_0 {
+	status = "okay";
 	#sound-dai-cells = <0>;
+	rockchip,clk-trcm = <1>;
 };
 
 &pdm_8ch {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
index 7932d0a8545d..8644d443f58c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
@@ -45,9 +45,14 @@
 	status = "disabled";
 };
 
-&dummy_codec {
-	status = "disabled";
+&bluetooth_sound {
+	status = "okay";
+};
+
+&i2s_2ch_0 {
+	status = "okay";
 	#sound-dai-cells = <0>;
+	rockchip,clk-trcm = <1>;
 };
 
 &pdm_8ch {
-- 
2.35.3

